File: 697. Degree of an Array/solution.py
class Solution(object):
  def findShortestSubArray(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    left = {}
    right = {}
    count = {}
    
    for index, num in enumerate(nums):
      if num not in left:
        left[num] = index
      right[num] = index
      if num not in count:
        count[num] = 1
      else:
        count[num] += 1
        
    maxCount = 0
    for num in count:
      if count[num] > maxCount:
        maxCount = count[num]
    
    result = float("inf")
    
    for num in count:
      if count[num] == maxCount:
        newResult = right[num] - left[num] + 1
        if newResult < result:
          result = newResult
          
    return result

File: 697. Degree of an Array/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var findShortestSubArray = function(nums) {
  let left = {};
  let right = {};
  let count = {};

  for (let index = 0; index < nums.length; index++){
    let num = nums[index];
    if (left[num] == undefined){
      left[num] = index;
    }
    right[num] = index;
    count[num] = (count[num] || 0) + 1;
  }

  let maxCount = 0;
  for (let num in count){
    if (count[num] > maxCount){
      maxCount = count[num];
    }
  }

  result = Infinity;

  for (let num in count){
    if (count[num] === maxCount){
      newResult = right[num] - left[num] + 1;
      if (newResult < result){
        result = newResult;
      }
    }
  }

  return result;
};